examples/sphere/solidbody.jl

```diff
@@ -5,112 +5,144 @@ import ClimaCore: slab, Fields, Domains, Topologies, Meshes, Spaces
+    domain = Domains.SphereDomain(R)
+    mesh = Meshes.Mesh2D(domain, Meshes.EquiangularSphereWarp(), ne)
+    grid_topology = Topologies.Grid2DTopology(mesh)
+    quad = Spaces.Quadratures.GLL{Nq}()
+    space = Spaces.SpectralElementSpace2D(grid_topology, quad)
+
+    coords = Fields.coordinate_field(space)
+
+    Δh[k] = 2 * R / ne
+
+    h_init = map(coords) do coord
+        n_coord = Geometry.components(Geometry.Cartesian123Point(coord))
+        n_center = Geometry.components(Geometry.Cartesian123Point(center))
+        # https://en.wikipedia.org/wiki/Great-circle_distance
+        rd = R * atan(norm(n_coord × n_center), dot(n_coord, n_center))
+
+        if test_name == gaussian_test_name
+            h0 * exp(-(rd / r0)^2 / 2)
+        else # default test case, cosine bell
+            if rd < r0
+                h0 / 2 * (1 + cospi(rd / r0))
+            else
+                0.0
+            end
+        end
     end
 
-end
-
-u = map(coords) do coord
-    ϕ = coord.lat
-    λ = coord.long
-
-    u = u0 * (cosd(α0) * cosd(ϕ) + sind(α0) * cosd(λ) * sind(ϕ))
-    v = -u0 * sind(α0) * sind(λ)
-    Geometry.UVVector(u, v)
-end
+    u = map(coords) do coord
+        ϕ = coord.lat
+        λ = coord.long
 
-function rhs!(dh, h, u, t)
-    wdiv = Operators.WeakDivergence()
-    grad = Operators.Gradient()
-    div = Operators.Divergence()
+        uu = u0 * (cosd(α0) * cosd(ϕ) + sind(α0) * cosd(λ) * sind(ϕ))
+        uv = -u0 * sind(α0) * sind(λ)
+        Geometry.UVVector(uu, uv)
+    end
 
-    # compute hyperviscosity first
-    @. dh = wdiv(grad(h))
-    Spaces.weighted_dss!(dh)
-    @. dh = -D₄ * wdiv(grad(dh))
+    function rhs!(dh, h, u, t)
+        div = Operators.Divergence()
 
-    # add in pieces
-    @. dh += -div(h * u)
-    Spaces.weighted_dss!(dh)
-end
-rhs!(similar(h_init), h_init, u, 0.0)
-
-# Solve the ODE operator
-T = 86400 * 12
-dt = 90 * 60
-prob = ODEProblem(rhs!, h_init, (0.0, T), u)
-sol = solve(
-    prob,
-    SSPRK33(),
-    dt = dt,
-    saveat = dt,
-    progress = true,
-    adaptive = false,
-    progress_message = (dt, u, p, t) -> t,
-)
-
-@info "Solution norm at T = 0: ", norm(h_init)
-@info "Solution norm at T = $(T): ", norm(sol.u[end])
-@info "L2 error at T = $(T): ", norm(sol.u[end] .- h_init)
-
-ENV["GKSwstype"] = "nul"
-import Plots
-Plots.GRBackend()
+        @. dh = -div(h * u) # strong form of equation
+        Spaces.weighted_dss!(dh)
+    end
 
-dirname = "cg_sphere_solidbody"
-path = joinpath(@__DIR__, "output", dirname)
-mkpath(path)
+    # Set the RHS function
+    rhs!(similar(h_init), h_init, u, 0.0)
+
+    # Solve the ODE
+    T = 86400 * 12
+    dt = 30 * 60
+    prob = ODEProblem(rhs!, h_init, (0.0, T), u)
+    sol = solve(
+        prob,
+        SSPRK33(),
+        dt = dt,
+        saveat = dt,
+        progress = true,
+        adaptive = false,
+        progress_message = (dt, u, p, t) -> t,
+    )
+    err[k] = norm(sol.u[end] .- h_init) / length(h_init)
```

Comment:
`h_init` will have unit length `1`, is this what you want to compute?

Reply:
You're right. I was looking at the same helper function in `test/fdspaces.jl` . I would think the relative error would just be `(exact - computed)/exact`, but in that case `length(parent(exact))` was used, so I got confused

Reply:
I'm going to leave it as it was, except for the `length(h_init)` bit (which wasn't going to have unit length, as we said above though. Now that I looked at it more closely)